test/mitmproxy/proxy/test_mode_servers.py

```diff
@@ -278,12 +278,11 @@ async def test_udp_start_error():
     with taddons.context():
         inst = ServerInstance.make("dns@127.0.0.1:0", manager)
         await inst.start()
-        port = inst.listen_addrs[0][1]
-        inst2 = ServerInstance.make(f"dns@127.0.0.1:{port}", manager)
+        port = inst.listen_addrs[1][1]
         with pytest.raises(
             Exception, match=f"Failed to bind UDP socket to 127.0.0.1:{port}"
         ):
-            await inst2.start()
+            await mitmproxy_rs.start_udp_server("127.0.0.1", port, MagicMock())
```

Comment:
Changes here can be reverted too, right?

Reply:
Using the DNS server here would raise a similar TCP exception first since we'd be starting the TCP server first in `listen()` but we're checking for UDP here

Reply:
Ah, I see. Thanks. Let's adjust the test to use `reverse:udp://127.0.0.1:1234@0` instead of dns then, this is really supposed to test ServerInstance and not mitmproxy_rs. :)

Reply:
I think you meant `reverse:udp@127.0.0.1:0`

Reply:
My bad, realised that `reverse:udp@127.0.0.1:0` and `reverse:udp://127.0.0.1:0` perform different things. Can you please explain the difference?